cmd/faucet/faucet.go

```diff
@@ -213,14 +213,21 @@ type faucet struct {
 	nonce    uint64             // Current pending nonce of the faucet
 	price    *big.Int           // Current gas price to issue funds with
 
-	conns    []*websocket.Conn    // Currently live websocket connections
+	conns    []*wsConn            // Currently live websocket connections
 	timeouts map[string]time.Time // History of users and their funding timeouts
 	reqs     []*request           // Currently pending funding requests
 	update   chan struct{}        // Channel to signal request updates
 
 	lock sync.RWMutex // Lock protecting the faucet's internals
 }
 
+// wsConn wraps a websocket connection with a write mutex as the underlying
+// websocket library does not synchronize access to the stream.
+type wsConn struct {
```

Comment:
We have a similar construct here: https://github.com/ethereum/go-ethereum/blob/master/ethstats/ethstats.go#L100  -- used for ethstats reporting. Would it make sense to use the same? 
The other one as a separate lock for reading -- isnt' that needed in this case?

Reply:
Not sure it's worth it.

- The duplicated code is a couple lines
- Exposing it in ethstats seems wrong, we'd need to move it into some common package,
- Ethstats wraps the write directly, but we need setdeadline too wrapped together, so we'd need lower level access.
- We only read in a dedicated goroutine here, so no need for read locking (though I think it's the same in ethstats).